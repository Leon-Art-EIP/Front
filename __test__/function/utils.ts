import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
