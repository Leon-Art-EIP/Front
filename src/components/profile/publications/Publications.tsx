import { ElementType } from "react";
import { IProfileArt } from "../../../interfaces/profile/profileCollection";

interface IPublicationsProps {
  profileArts: IProfileArt[];
  link: ElementType<{ children: JSX.Element; href: string }>;
}

export default function Publications({ link: Link, ...props }: IPublicationsProps): JSX.Element {
  if (props.profileArts.length === 0) {
    return <div className="text-2xl">Aucune publication</div>;
  }

  return (
    <div className="w-full flex justify-center bg-background-hl p-4 rounded">
      <div className="grid xl:grid-cols-4 md:grid-cols-3 grid-cols-2 gap-4 mx-auto">
        {props.profileArts.map((picture) => (
          <Link
            key={`art-${picture.id}`}
            href={`/single/${picture.id}`}
            className="2xl:w-64 2xl:h-64 xl:w-56 xl:h-56 md:w-44 md:h-44 w-36 h-36 rounded-xl overflow-hidden hover:scale-[1.03] hover:opacity-75 duration-300 hover:shadow-2xl transition ease-in-out cursor-pointer"
          >
            <img src={picture.src} alt="art" className="w-full h-full object-cover object-center" />
          </Link>
        ))}
      </div>
    </div>
  );
}
