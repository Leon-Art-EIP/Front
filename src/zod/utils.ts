export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
