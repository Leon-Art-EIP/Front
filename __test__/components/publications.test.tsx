import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import Publications from "../../src/components/profile/publications/Publications";
import Link from "../../src/components/link/Link";
import leonart from "../../public/leonart_logo.png";

test("Publications snapshot", () => {
  testSnapshot(
    <Publications
      link={Link}
      profileArts={[
        {
          id: "1",
          src: leonart,
        },
      ]}
    />
  );
});
