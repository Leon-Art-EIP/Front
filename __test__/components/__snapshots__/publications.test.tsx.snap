// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Publications snapshot 1`] = `
<div
  className="w-full flex flex-col bg-background-hl p-4 rounded gap-4"
>
  <div
    className="flex flex-wrap gap-4"
  >
    <a
      className="2xl:w-64 2xl:h-64 xl:w-56 xl:h-56 md:w-44 md:h-44 w-36 h-36 rounded-xl overflow-hidden hover:scale-[1.03] hover:opacity-75 duration-300 hover:shadow-2xl transition ease-in-out cursor-pointer"
      href="/single/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <img
        alt="art"
        className="w-full h-full object-cover object-center"
        src="/public/leonart_logo.png"
      />
    </a>
  </div>
</div>
`;
